### DISCUSSION TITLE
How can i read json file in my backend serverless

### DISCUSSION THREAD

**Discussion by OrYehudaBenShimol** [vercel/community#2970] [Help] [answered]
### Summary

Hi everyone,

After deploying my web app to vercel I've encountered a problem which wont let me read a json file  which is located in the backend repository.

This is my implementation:

const fs = require('fs');
const path = require('path');
// Get all statistics data from json file.
const getAllData = async (req,res) =>{
         res.setHeader('Access-Control-Allow-Origin', '*');
         res.setHeader('Access-Control-Allow-Methods', 'OPTIONS, GET, POST, PUT, DELETE');
         res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
         const file = path.join(process.cwd(), 'files', 'morning-meeting.json');
         const stringified = fs.readFileSync(file, 'utf8');
         const tempData = "[" + stringified + "]"
         const jsonData =  JSON.parse(tempData)
         res.status(200).json(jsonData);
};  

module.exports = {
    getAllData,
}

The file I'm trying to read is "morning-meeting.json" and its path is in the home folder of my git repository.

Thanks in advance.

### Steps to Reproduce

this is the error message i get in the backend logs:

/api/statistics GET
Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: ENOENT: no such file or directory, open '/var/task/files/morning-meeting.json'","reason":{"errorType":"Error","errorMessage":"ENOENT: no such file or directory, open '/var/task/files/morning-meeting.json'","code":"ENOENT","errno":-2,"syscall":"open","path":"/var/task/files/morning-meeting.json","stack":["Error: ENOENT: no such file or directory, open '/var/task/files/morning-meeting.json'","    at Object.openSync (node:fs:601:3)","    at Object.readFileSync (node:fs:469:35)","    at getAllData (/var/task/controllers/statisticsController.js:33:33)","    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)","    at next (/var/task/node_modules/express/lib/router/route.js:144:13)","    at Route.dispatch (/var/task/node_modules/express/lib/router/route.js:114:3)","    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)","    at /var/task/node_modules/express/lib/router/index.js:284:15","    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:346:12)","    at next (/var/task/node_modules/express/lib/router/index.js:280:10)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: ENOENT: no such file or directory, open '/var/task/files/morning-meeting.json'","    at process.<anonymous> (file:///var/runtime/index.mjs:1189:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown

**Comment by OrYehudaBenShimol** [comment, marked as answer]
I had a problem in the path.join
just removed the 2nd parameter and it worked

**Reply by FahadSalamatRandhawa** [reply]
I am facing same issue, I tried your approach but its not working
"pnpm run build" gives error, the code works fine locally(without this specific API, the other api's read the file ok on localhost but error when deployed to vercel)

Link to issue : https://github.com/orgs/vercel/discussions/3763

![image](https://github.com/vercel/community/assets/66182071/33720f32-5d22-4b79-911a-383979451588)
